tests/e2e/performance/test_small_file_workload.py

```diff
@@ -105,152 +235,51 @@ def add_key(self, key, value):
+        Aggregation results from all hosts in single sample
 
         Args:
-            res (dict) : dictionary of results
+            host_res (dict): dictionary of host results - user for multi hosts
+                             test
 
         Returns:
-            dict : dictionary with the aggregate results.
+            dict: dictionary of all the host results aggregated with the host
+                  results that provide as argument to one set of results
 
         """
-
         results = {}
-        for key in self.managed_keys.keys():
-            if self.managed_keys[key]['name'] in res.keys():
-                results[key] = self.managed_keys[key]['op'](
-                    res[self.managed_keys[key]['name']]
-                )
-
-        # This is the place to check in host (treads) deviation.
-
-        return results
-
-    def combine_results(self, results, clear):
-        """
-        Combine 2 or more results (hosts in sample / samples in test)
-        to one result.
-
-        Args:
-            results (dict): dictionary of results to combine
-            clear (bool): return only combined results or not.
-                          True - return only combined results
-                          False - add the combine results to originals results
-
-        Returns:
-            dict : dictionary of results records
-
-        """
-
-        res = {}
-        log.info(f'The results to combine {results}')
-        for rec in results.keys():
-            record = results[rec]
-            for key in self.managed_keys.keys():
-                # not all operation have all values, so i am using try
-                try:
-                    val = float('{:.2f}'.format(record[key]))
-                    if self.managed_keys[key]['name'] in res.keys():
-                        res[self.managed_keys[key]['name']].append(val)
-                    else:
-                        res[self.managed_keys[key]['name']] = [val]
-                except Exception:
-                    pass
-        if not clear:
-            res.update(self.aggregate_threads_results(res))
+        log.info('Aggregating hosts results to one test-results')
+        if self.results['full-res'] == {}:
+            self.results['full-res'] = host_res
+            log.info('This is the first pod - return results as is')
         else:
-            res = self.aggregate_threads_results(res)
-        return res
-
-    def aggregate_host_results(self):
-        """
-        Aggregation results from all hosts in single sample
-
-        """
-
-        results = {}
-
-        for op in self.results['operations']:
-            for smp in range(self.results['samples']):
-                sample = smp + 1
-                if op in self.results['full-res'].keys():
-                    self.results['full-res'][op][sample] = self.combine_results(
-                        self.results['full-res'][op][sample], True)
+            for op in self.results['operations']:
```

Comment:
Can you please add docstring explaining the process here?

Reply:
done